runelite-client/src/main/java/net/runelite/client/plugins/mta/AlchemyRoom.java

```diff
@@ -0,0 +1,394 @@
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.mta;
+
+import java.awt.Color;
+import java.awt.Graphics2D;
+import java.awt.image.BufferedImage;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import javax.inject.Inject;
+import com.google.common.eventbus.Subscribe;
+import net.runelite.api.ChatMessageType;
+import net.runelite.api.Client;
+import net.runelite.api.GameObject;
+import net.runelite.api.ItemComposition;
+import net.runelite.api.ObjectID;
+import net.runelite.api.Perspective;
+import net.runelite.api.Point;
+import net.runelite.api.coords.WorldPoint;
+import net.runelite.api.events.ChatMessage;
+import net.runelite.api.events.GameObjectDespawned;
+import net.runelite.api.events.GameObjectSpawned;
+import net.runelite.api.events.GameTick;
+import net.runelite.api.widgets.Widget;
+import net.runelite.api.widgets.WidgetID;
+import net.runelite.api.widgets.WidgetInfo;
+import net.runelite.api.widgets.WidgetItem;
+import net.runelite.client.game.ItemManager;
+import net.runelite.client.ui.overlay.infobox.InfoBoxManager;
+
+public class AlchemyRoom extends MTARoom
+{
+
+	private static final int TEXT_Z_OFFSET = 180;
+	private static final int IMAGE_Z_OFFSET = 150;
+	private static final int CAPACITY = 8;
+	private static final int INFO_START = 5;
+	private static final int BEST_POINTS = 30;
+
+	private static final List<Integer> CUPBOARDS = Arrays.asList(ObjectID.CUPBOARD_23678, ObjectID.CUPBOARD_23680,
+			ObjectID.CUPBOARD_23682, ObjectID.CUPBOARD_23684, ObjectID.CUPBOARD_23686, ObjectID.CUPBOARD_23688,
+			ObjectID.CUPBOARD_23690, ObjectID.CUPBOARD_23692);
+	private static final String YOU_FOUND = "You found:";
+	private static final String EMPTY = "The cupboard is empty.";
+
+	private final HashMap<AlchemyItem, BufferedImage> IMAGE_MAP = new HashMap<>();
+	private final AlchemyItem[] locations = new AlchemyItem[CAPACITY];
+	private final List<GameObject> cupboards = new ArrayList<>(CAPACITY);
+	private final MTAPlugin plugin;
+	@Inject
```

Comment:
I would add empty lines separating some of these variables, not everyone in this project does it, and is not required for the code style, but makes the code easier to read

Reply:
Grouped variables by initialization time